File: whois/get_axfr.py
#!/usr/bin/python
import dns.query
import dns.zone
import time

def get_axfr(zone_name):
	axfr_query = dns.query.xfr('127.0.0.1', zone_name)
	try:
		zones = dns.zone.from_xfr(axfr_query)
	except dns.exception.FormError:
		print "Format error...."
		return False

	output = "; Zone data for %s\n" % zone_name
	output += "; Generated @ %s\n" % time.strftime("%d/%m/%Y %H:%M")
	output += "; %d records\n" % len(zones.nodes.keys())

	for record in zones.nodes.keys():
		output += zones.nodes[record].to_text(record)
		output += "\n"

	return output

print get_axfr('damian.internal')

File: whois/get_axfr.pl
#!/usr/bin/env perl
use strict;
use Net::DNS;

my $res  = Net::DNS::Resolver->new;
$res->nameservers("127.0.0.1");

my @zone = $res->axfr("damian.internal");
if (@zone) {
	my $zone_output = "";

	$zone_output .= "; Generated at " . localtime . "\n";
	foreach my $rr (@zone) {
		my $record = $rr->string;
		$zone_output .= $record . "\n";
	}

	print $zone_output;
} else {
	print 'Zone transfer failed: ', $res->errorstring, "\n";
}
